### PR TITLE
docs: Installation guide supported platforms

### PR DESCRIPTION
**Pull request by ibuziuk**
## What does this pull request change

- bump minimal kube version to 1.21 (used in OpenShift 4.8)
- remove the installation methods from the supported platforms
- add a warning about Users' Authentication setup req for vanilla k8s
- list cloud providers alphabetically

 Installation guide supported platforms

image

## What issues does this pull request fix or reference

part of https://github.com/eclipse/che/issues/21284

## Pull request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [x] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)
- [x] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [x] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

File: modules/administration-guide/examples/ref_che-supported-platforms-and-installation-methods.adoc [hunk 1]
```diff
@@ -34,27 +31,27 @@ a|* `minikube`
 |OperatorHub
 |===
 
+WARNING: Setting up an OIDC provider is required for deploying {prod-short} on {kubernetes} infrastructures. For {ocp} no additional setup is needed.
```
Comment (l0rd) [inline modules/administration-guide/examples/ref_che-supported-platforms-and-installation-methods.adoc]:
```suggestion
WARNING: Setting up [users authentication](https://kubernetes.io/docs/reference/access-authn-authz/authentication/) is required for deploying {prod-short} on {kubernetes} infrastructures. For {ocp} no additional setup is needed.
```

Comment (l0rd) [reply modules/administration-guide/examples/ref_che-supported-platforms-and-installation-methods.adoc]: Reading it the first time I thought the warning was about `Table 2. Local installation on OpenShift` (that comes just before). So I would say that it should be moved right after `Table 3. Public cloud installation on Kubernetes or OpenShift`, but I am not an expert. Take this comment with a grain of salt.

**Review by l0rd — COMMENTED** [review]
Something that bothers me with current doc is that how to install Che (with `chectl`, `OperatorHub` or even `helm`) should not be covered here. The details here are not accurate (`chectl` is recommended if the admin is more comfortable with the command line or wants to setup automation, it's not a backup installation method. And the OperatorHub can be used on any kube cluster that has `olm`, including minikube, not just on OCP. The helm chart is not mentioned) and it's already covered in the Installing Che section.

**Comment by ibuziuk** [comment]
@l0rd thanks for the review - sounds reasonable, will remove the installation methods from the supported platforms section

File: modules/administration-guide/examples/ref_che-supported-platforms-and-installation-methods.adoc [hunk 2]
```diff
@@ -34,27 +31,27 @@ a|* `minikube`
 |OperatorHub
 |===
 
+WARNING: Setting up [users authentication](https://kubernetes.io/docs/reference/access-authn-authz/authentication/) is required for deploying {prod-short} on {kubernetes} infrastructures. For {ocp} no additional setup is needed.
```
Comment (swatikarmarkar) [inline modules/administration-guide/examples/ref_che-supported-platforms-and-installation-methods.adoc]:
```suggestion
WARNING: Setting up [User's Authentication](https://kubernetes.io/docs/reference/access-authn-authz/authentication/) is required for deploying {prod-short} on {kubernetes} infrastructures. For {ocp} no additional setup is needed.
```

Comment (ibuziuk) [reply modules/administration-guide/examples/ref_che-supported-platforms-and-installation-methods.adoc]: @swatikarmarkar maybe `Users' Authentication` in this case?

**Comment by ibuziuk** [comment]
@l0rd PR has been updated for the final review

**Review by swatikarmarkar — APPROVED** [review]
The changes look good.

File: modules/administration-guide/examples/ref_che-supported-platforms-and-installation-methods.adoc [hunk 3]
```diff
@@ -2,63 +2,27 @@
...
 * Google Cloud
 * IBM Cloud
-.^|`{prod-cli}` using the Operator
-.^|`{prod-cli}` using the Operator
-.^|OperatorHub
-|===
+* Microsoft Azure
+* Rancher
+
+WARNING: Setting up [Users' Authentication](https://kubernetes.io/docs/reference/access-authn-authz/authentication/) is required for deploying {prod-short} on {kubernetes} infrastructures. For {ocp} no additional setup is needed.
```
Comment (l0rd) [inline modules/administration-guide/examples/ref_che-supported-platforms-and-installation-methods.adoc]: You are using markdown links, you should use asciidoc instead.

Comment (ibuziuk) [reply modules/administration-guide/examples/ref_che-supported-platforms-and-installation-methods.adoc]: ack, used what was suggested in https://github.com/eclipse-che/che-docs/pull/2284#discussion_r849692962 ;-)

Comment (ibuziuk) [reply modules/administration-guide/examples/ref_che-supported-platforms-and-installation-methods.adoc]: fixed

Comment (l0rd) [reply modules/administration-guide/examples/ref_che-supported-platforms-and-installation-methods.adoc]: oh sorry, my fault 😄

**Review by l0rd — APPROVED** [review]
👍 this article is in a much better shape now.

**Pull request merged**